primitives/arithmetic/src/fixed_point.rs

```diff
@@ -670,35 +679,38 @@ macro_rules! implement_fixed {
 
 			#[test]
 			fn op_neg_works() {
-				let a = $name::saturating_from_integer(5);
+				let a = $name::zero();
 				let b = -a;
 
-				// Positive.
-				assert_eq!($name::saturating_from_integer(-5), b);
+				// Zero.
+				assert_eq!(a, b);
 
-				let a = $name::saturating_from_integer(-5);
-				let b = -a;
+				if $name::min_value() != $name::zero() {
```

Comment:
```suggestion
				if $name::min_value() != $name::zero() {
```

do you think ` $name::min_value() != $name::zero()` could always be the answer to `is_signed`? (regarding my other comment of changing the behaviour of neg() if the type is unsigned.)